### PR TITLE
Update old Transformation Lifecycle and Plugin docs

### PR DESCRIPTION
**Pull request by c-mueller**

### CODE REVIEW

File: docs/dev/core/plugin-guidelines.md [hunk 1]
```diff
@@ -1,54 +1,46 @@
...
 
 We use Springs Annotation based dependency injection in order to load the Plugins. A class that implements or extends the interface / class listed above also needs a `@Component` annotation to let Spring autoconstruct a instance. This also means that the class needs a public constructor with no arguments, Autowiring other dependencies is possible but it is not required for any operations between plugin and core. For any Interactions between these the given operation context (`TransformationContext` for transformations)  
 
 Currently Spring Boot is setup to only look for `@Component` annotations on classes in the package `org.opentosca.toscana` and it's child packages. Therefore the main plugin class has to be located within this package hierarchy.
 
-## The `AbstractPlugin` class
+## The `ToscanaPlugin` class
 
-The following section describes the implementation of a plugin using the abstract class `AbstractPlugin`. 
+The following section describes the implementation of a plugin using the abstract class `AbstractPlugin`.
```
Comment (mmunozba) [inline docs/dev/core/plugin-guidelines.md]: AbstractPlugin -> ToscanaPlugin

File: docs/dev/core/plugin-guidelines.md [hunk 2]
```diff
@@ -1,54 +1,46 @@
...
-### Required Implementations
+Since we use the TLI (TransformationLifecycleInterface) now for every plugin, the original possiblility to implement a low level plugin has been removed, you have to implemtent a Plugin based on the TLI for that you will have to implement the `ToscanaPlugin<T>` class where T is your classname of your Lifecycle implementaion.
 
-The following two methods have to be implemented in order to contruct a corresponding `Platform` object.
-```Java
-public abstract String getName();
-public abstract String getIdentifier();
-```
-The `getName()` method returns the Display name of the plugin and the `getIdentifier()` method returns the unique identifier (matching the regular expression `[a-z_-]+`). If a plugin with the same identifier is already registered launching of the application will not continue and the application will stop (throwing a `IllegalArgumentException`)
+We use a factory method (`getInstance()`) to construct a instance of your Lifecycle implementation for every transformation that gets performed.
```
Comment (mmunozba) [inline docs/dev/core/plugin-guidelines.md]: a instance -> an instance

File: docs/dev/core/plugin-guidelines.md [hunk 3]
```diff
@@ -1,54 +1,46 @@
...
 Currently Spring Boot is setup to only look for `@Component` annotations on classes in the package `org.opentosca.toscana` and it's child packages. Therefore the main plugin class has to be located within this package hierarchy.
 
-## The `AbstractPlugin` class
+## The `ToscanaPlugin` class
 
-The following section describes the implementation of a plugin using the abstract class `AbstractPlugin`. 
+The following section describes the implementation of a plugin using the abstract class `AbstractPlugin`.
 
-### Required Implementations
+Since we use the TLI (TransformationLifecycleInterface) now for every plugin, the original possiblility to implement a low level plugin has been removed, you have to implemtent a Plugin based on the TLI for that you will have to implement the `ToscanaPlugin<T>` class where T is your classname of your Lifecycle implementaion.
```
Comment (mmunozba) [inline docs/dev/core/plugin-guidelines.md]: possiblility -> possibility

File: docs/dev/core/plugin-guidelines.md [hunk 4]
```diff
@@ -1,54 +1,46 @@
...
 
 #### Logging
 
 Because we want to be able to only log the events for a specific transformation in a special file and by the REST API a special logger factory has to be used if you want to log while the transformation is running. Using this feature is highly recommended because we currently don't have any other option that tells the user whats currently going on with the transformation.
 
 #### Handling Exceptions
 
-You have to handle exceptions within the `transform(...)` method. unless the given exception should result in the abortion of the transformation process. If a exception gets thrown out of the method it will be caught and the execution of the transformation will be stopped immediately and the transformation state gets set to `ERROR`.
+Because the phase methods of the TransformationLifecycle do not have a `throws` declaration every exeption thats not a `RuntimeException` will have to get caught. If you want the transformation to fail if a exception gets thrown you should rethrow the exception, wrapped in a `TransformationFailiureException` (extends `RuntimeException`)
```
Comment (mmunozba) [inline docs/dev/core/plugin-guidelines.md]: `...to fail if a exception gets...` -> `...to fail if an exception gets...`

File: docs/dev/plugins/target-artifact-conventions/target-artifact-conventions.md [hunk 5]
```diff
@@ -37,6 +37,8 @@ The purpose of this document is to specify conventions for the target artifact t
 As mentioned before our target artifact folder has a `scripts` folder.
 To prevent random names scripts also have to follow conventions.
 
+It is allowed to put some (important) scripts into the `output/` directory instead of the `output/scripts/` directory. This currently gets done for deployment scripts (`deploy.sh`)
```
Comment (mmunozba) [inline docs/dev/plugins/target-artifact-conventions/target-artifact-conventions.md]: For the CloudFormation plugin, the `deploy.sh` still gets put in the `output/scripts/` directory, should this be changed?

File: docs/dev/core/plugin-guidelines.md [hunk 6]
```diff
@@ -1,54 +1,46 @@
-# Plugin Developement Guidelines
-
-Transformation Plugins for the TOSCAna transformer have to implement the `TransformationPlugin` interface. However it is recommended to extend the abstract class `AbstractPlugin`.
-
 ## Plugin Loading
 
 We use Springs Annotation based dependency injection in order to load the Plugins. A class that implements or extends the interface / class listed above also needs a `@Component` annotation to let Spring autoconstruct a instance. This also means that the class needs a public constructor with no arguments, Autowiring other dependencies is possible but it is not required for any operations between plugin and core. For any Interactions between these the given operation context (`TransformationContext` for transformations)
```
Comment (hnicke) [inline docs/dev/core/plugin-guidelines.md]: The section ends abruptly.

File: docs/dev/plugins/target-artifact-conventions/target-artifact-conventions.md (hunk 5 above)
Comment (nfode) [reply docs/dev/plugins/target-artifact-conventions/target-artifact-conventions.md]: I also thought that scripts go into the `output/scripts/` directory.

Comment (c-mueller) [reply docs/dev/plugins/target-artifact-conventions/target-artifact-conventions.md]: this was "problem" from my side, did fix it at its origin. This part has also been removed

**Pull request merged**